datajoint/s3.py

```diff
@@ -150,3 +189,64 @@ def delete(self, rpath):
             raise DataJointError(
                 'error deleting file {r} ({e})'.format(r=rpath, e=e)
             )
+
+
+class S3FileHandler(ExternalFileHandler):
+
+    required = ('bucket', 'location', 'aws_access_key_id',
+                'aws_secret_access_key',)
+
+    def __init__(self, store, database):
+        super().__init__(store, database)
+
+        self.check_required(store, 's3', S3FileHandler.required)
+
+        self._bucket = bucket(
+            aws_access_key_id=self._spec['aws_access_key_id'],
+            aws_secret_access_key=self._spec['aws_secret_access_key'],
+            bucket=self._spec['bucket'])
+
+        self._location = self._spec['location']
+
+    def make_path(self, hash):
+        '''
+        create a remote s3 path for the given hash.
+
+        S3 has no path.join; '' is root, '/' is delimiter.
+
+        Leading slashes will create an empty-named top-level folder.
+        therefore we leading slashes from location and do not create them.
+
+        This is done here and not in the Bucket class since it is not
+        technically 'illegal' and Bucket encapsulates low-level access.
+        '''
+        if(len(self._location) == 0 or self._location == '/'):
+            return hash
+        if self._location[0] == '/':
+            return self._location[1:] + '/' + hash
+        else:
+            return self._location + '/' + hash
+
+    def put(self, obj):
+        (blob, hash) = self.hash_obj(obj)
+        rpath = self.make_path(hash)
+        self._bucket.put(BytesIO(blob), rpath)
+        return (blob, hash)
+
+    def get_blob(self, hash):
+        rpath = self.make_path(hash)
```

Comment:
why `make_path` in `get_blob`?  Shouldn't you return an error if it's not there?